docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,189 @@
+
+### Collecting LDAP Members: Searching for one Group
+An `LDAPMemberLister` will be used inside of the sync job to do the first step in the sync: locate the LDAP entries that correspond to the group being synced, given the filter for that group:
+
+```go
+// LDAPMemberLister lists the LDAP entries that correspond to the members of the given Group
+type LDAPMemberLister interface {
+  ListMembers(groupFilter string) (members *[]ldap.Entry, err error)
+}
+```
+#### Given Members as First-Class LDAP Entries
+If members are first-class LDAP entries, the `LDAPFirstClassMemberLister` can simply query the LDAP server for members of a group and return those entries.
+
+#### Given Members as LDAP Annotations on LDAP Groups
+If LDAP group members are returned by reference in an LDAP entry attribute, another LDAP filter will need to be created with the given information in order to find the LDAP first-class entry for the user, if it exists. The generation of the LDAP query from the description of the LDAP group member will need to be specified by the user requesting a sync.
+
+### Mapping LDAP Members to OpenShift `Identities`
+This will be done in a similar way to the mapping for [LDAP password processing](https://github.com/openshift/origin/blob/master/pkg/auth/authenticator/password/ldappassword/ldap.go#L227-243) once the requisite data is collected.
+
+### Mapping OpenShift `Identities` Representing LDAP Members to OpenShift `Users`
+A new `UserIdentityMapper` will need to be created: the `ForcedDeterministicUserIdentityMapper`, which will assume that the unique identifier retrieved from the LDAP response will be directly equal to the unique identifier (`uid`) of the OpenShift `User` that matches this `Identity`. This is required as otherwise there is no deterministic mapping from the `Identity` created and any OpenShift `User`. This will need to be boldly documented.
+
+## Internalizing LDAP Responses: Building OpenShift `Groups`
+### Building OpenShift `Groups` to Hold OpenShift `Users` Created or Referenced by LDAP Responses
+Given the identifying filter uesd to find a LDAP group, the corresponding OpenShift `Group` can either be found using the `externalRecordProvider` `Label` and `LDAPFilterGenerator` or created it it does not yet exist. If the group returned from the LDAP server is a first-class entry, information can be migrated to the OpenShift `Group`, but if the group is encode as attributes of first-class LDAP user entries, no information other than the name of the group can be carried over. In either case, data from the old OpenShift `Group` is kept unless overridden by the update.
```

Comment:
Be more specific.  Non-conflicting `ObjectMeta` can remain, but `Users` will always be fully replaced.